app/404/page.tsx

```diff
@@ -0,0 +1,14 @@
+import { useTranslations } from 'next-intl';
+import styles from './page.module.css';
+import { CustomLink } from '../components/CustomLink/CustomLink';
+import { RoutePath } from '@/utils/utils';
+
+export default function NotFoundPage() {
+  const t = useTranslations('404');
```

Comment:
Looks like all this string keys which are used for getting `translate` function can be stored in the separate file. It will increase reusability.